src/pfe/portal/modules/User.js

```diff
@@ -183,21 +184,46 @@ module.exports = class User {
   /**
    * Function to cancel load on project
    */
-  async cancelLoad(project) {
+  cancelLoad(project) {
     log.debug("cancelLoad: project " + project.projectID + " loadInProgress=" + project.loadInProgress);
     this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelling' });
 
     if (project.loadInProgress) {
-      project.loadInProgress = false;
       log.debug("Cancelling load for config: " + JSON.stringify(project.loadConfig));
-      let cancelLoadResp = await this.loadRunner.cancelRunLoad(project.loadConfig);
-      this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelled' });
-      return cancelLoadResp;
+      this.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelling' });
+      const that = this;
+      return promiseRetry({retries: 15, minTimeout: 2000, maxTimeout: 2000}, function (retry, number) {
+        log.info(`Attempting to cancel load for project ${project.projectID}. Attempt ${number}`);
+        return that.callCancelRunLoad(project)
+          .catch(function (err) {
+            retry(err);
+          });
+      })
+        .then(function (value) {
+          project.loadInProgress = false;
+          that.uiSocket.emit('runloadStatusChanged', { projectID: project.projectID, status: 'cancelled' });
```

Comment:
Instead of doing `const that = this` I think you should be able use arrow functions to pass the `this` context down:
```
return cancelLoadResp;	      return promiseRetry({retries: 15, minTimeout: 2000, maxTimeout: 2000}, (retry, number) => {
    log.info(`Attempting to cancel load for project ${project.projectID}. Attempt ${number}`);
    return that.callCancelRunLoad(project)
      .catch(function (err) {
        retry(err);
    });
})
```

Reply:
Also curious why you chose `promise-retry` over `async-retry` (which I notice is in the `package.json`) when `async-retry` means we can do `async/await` rather than conventional promises